charms/istio-pilot/src/istioctl.py

```diff
@@ -11,34 +12,50 @@ class IstioctlError(Exception):
 
 class Istioctl:
     def __init__(
-        self, istioctl_path: str, namespace: str = "istio-system", profile: str = "minimal"
+        self,
+        istioctl_path: str,
+        namespace: str = "istio-system",
+        profile: str = "minimal",
+        istioctl_extra_flags: Optional[List[str]] = [],
     ):
         """Wrapper for the istioctl binary.
 
         This class provides a python API for the istioctl binary, supporting install, upgrade,
         and other istioctl commands.
 
         Args:
+            istioctl_extra_flags (optional, list): A list containing extra flags to pass to istioctl
             istioctl_path (str): Path to the istioctl binary to be wrapped
             namespace (str): The namespace to install Istio into
             profile (str): The Istio profile to use for installation or upgrades
```

Comment:
Can we move this to the bottom of the list to keep it in the same order as the flags?

```suggestion
            istioctl_path (str): Path to the istioctl binary to be wrapped
            namespace (str): The namespace to install Istio into
            profile (str): The Istio profile to use for installation or upgrades
            istioctl_extra_flags (optional, list): A list containing extra flags to pass to istioctl
```

Reply:
Oh, I kept alphabetical order here, but sure, I will change it.